### PR TITLE
src: tidy-up conditions for CA bundle search

### PR DESCRIPTION
**Pull request by vszakats**
- delete redundant Schannel check.

- move `feature_ssl` check one level up from `FindWin32CACert()`.

- check `feature_ssl` early to skip a bunch of CA bundle search logic
  for no-ssl configurations.

---

w/o whitespace: https://github.com/curl/curl/pull/14841/files?w=1

### CODE REVIEW

**Comment by vszakats** [comment]
There is similar code in `lib/url.c` that limits CA bundle handling to Schannel _and_ SecureTransport:
https://github.com/curl/curl/blob/813995bb17da4a41c32235feeab156bf1d5bc059/lib/url.c#L427-L435

Is there a reason for these to be different?

**Comment by bagder** [comment]
> Is there a reason for these to be different than the condition in src/tool_operate.c?

Those are the two TLS backends that don't use a CA file but the native CA store by default.

**Comment by vszakats** [comment]
In that case wouldn't it be better to exclude SecureTransport in the command-line tool as well from the CA bundle detection?

(There is no issue per se, but it's unnecessary work AFAIU.)

**Comment by bagder** [comment]
Yes it would!

File: src/tool_operate.c [hunk 1]
```diff
@@ -3042,11 +3042,13 @@ static CURLcode transfer_per_config(struct GlobalConfig *global,
      * no environment-specified filename is found then check for CA bundle
      * default filename curl-ca-bundle.crt in the user's PATH.
      *
-     * If Schannel is the selected SSL backend then these locations are
-     * ignored. We allow setting CA location for schannel only when explicitly
-     * specified by the user via CURLOPT_CAINFO / --cacert.
+     * If Schannel or SecureTransport is the selected SSL backend then these
+     * locations are ignored. We allow setting CA location for Schannel and
+     * SecureTransport when explicitly specified by the user via
```
Comment (bagder) [inline src/tool_operate.c:3047]: nit: We usually refer to it as **Secure Transport** in documentation, using two words

Comment (vszakats) [reply src/tool_operate.c:3047]: Yes, I'm going to fix it in the tidy-up draft PR #14692. (also in url.c where this copy came from)

Comment (vszakats) [reply src/tool_operate.c]: @jay asked the same question when the `url.c` patch was submitted:
https://github.com/curl/curl/pull/12704#issuecomment-1891541873

If we don't mind making this compatibility break, it'd be more consistent to disregard these envvars by default for Secure Transport build. Apple's, Homebrew, and macport's (if I'm reading [it](https://github.com/macports/macports-ports/blob/master/net/curl/Portfile) correctly) curl are not using Secure Transport by default.

Comment (vszakats) [reply src/tool_operate.c]: All in all it doesn't matter given that Secure Transport is going to be deleted next May.

File: src/tool_doswin.c [hunk 2]
```diff
@@ -616,40 +616,29 @@ char **__crt0_glob_function(char *arg)
  */
```
Comment (jay) [inline src/tool_doswin.c:617]:
```suggestion
/ * Search and set the CA cert file for Windows.
 *
 * Do not call this function if Schannel is the selected SSL backend. We allow
 * setting CA location for Schannel only when explicitly specified by the user
 * via CURLOPT_CAINFO / --cacert.
 */
```
```

**Review by jay — APPROVED** [review]
I think this is fine however for posterity I suggest add a comment above the function that it should not be called for Schannel since it is not obvious now that the check is out of the function.

**Pull request closed without merging**